### DISCUSSION TITLE
Routing - How would you

### DISCUSSION THREAD

**Discussion by TheYetiNet** [mikecao/flight#570] [Q&A]
Hey Guys,

Firstly, want to say thanks this framework has covered so many projects for me and its great to see it moving to V3.

My new project is going to be a rather large API with a lot of end points. Currently there are a few ways I can think to do this using Routes in Flight.

1) Hard code the routes ( 100+ OUCH )
2) Wildcard them ( /AdminApi/* ect.. Deal with in controller, Calling either a Sub Controller / Model to do the dirty. )

Does anyone have a better suggestion ?

**Comment by TheYetiNet** [comment]
Been playing today and managed to get the following to work. Id be interested in feedback or alternatives still, just thought i'd share incase it helps

`Flight::route('/@controller/@action', function($controller, $action){
    $controllerClassName = 'Framework\\'. ucfirst($controller) . '\Controller\\'. ucfirst($controller) . 'Controller';

    $actionMethodName = ucfirst($action) . 'Action';

    if(class_exists($controllerClassName)) {
        $controllerInstance = new $controllerClassName();
        if(method_exists($controllerInstance, $actionMethodName)) {
            $controllerInstance->$actionMethodName();
        } else {
            Flight::notFound();
        }
    } else {
        Flight::notFound();
    }
});`

**Reply by n0nag0n** [reply]
So this is kinda where my initial thoughts lead to. Something like `Flight::route('/url/@controller/@method', 'Controller->methodRouter');` and then you would build a mini router inside the controller to help with that. You could (should) put it inside a `Flight::group()` to help with that but it will make it so that all your methods in the controller must be `public function someAction($controller, $method, $other_param)`. I have mulled over a custom alias inside of routes for `@controller` and `@method` so that you can have fewer routes to sort through (and I've done this successfully on another framework internally). The problem comes when you cannot use aliasing with it.

**Reply by TheYetiNet** [reply]
Thank you for that suggestion i will investigate if i can make that work. Difficult process to try and plan for early
